File: backend/src/main/java/me/performancereservation/domain/settlement/dto/SettlementUpdateResponse.java
package me.performancereservation.domain.settlement.dto;

import jakarta.validation.constraints.NotNull;
import me.performancereservation.domain.settlement.Settlement;

public record SettlementUpdateResponse (
        Long settlementId,
        Long performanceId,
        String bank,
        String account
) {
    public static SettlementUpdateResponse fromSettlement(Settlement settlement) {
        return new SettlementUpdateResponse(
                settlement.getId(),
                settlement.getPerformanceId(),
                settlement.getBank(),
                settlement.getAccount()
        );
    }
}

File: backend/src/main/kotlin/me/performancereservation/domain/settlement/dto/SettlementUpdateResponse.kt
package me.performancereservation.domain.settlement.dto

import me.performancereservation.domain.settlement.Settlement

data class SettlementUpdateResponse(
    val settlementId: Long,
    val performanceId: Long,
    val bank: String,
    val account: String
) {
    companion object {
        @JvmStatic
        fun fromSettlement(settlement: Settlement): SettlementUpdateResponse {
            return SettlementUpdateResponse(
                settlement.id!!,
                settlement.performanceId,
                settlement.bank,
                settlement.account
            )
        }
    }
}
